File: Semaphore_sharp/Semaphore_sharp/Customer.cs
using System;
using System.Threading;

namespace Semaphore_sharp
{
    public class Customer
    {
        private int _itemsCount;
        private int _pickDelay_ms;
        private Storage _storage;

        public Customer(int itemsCount, int pickDelay_ms, Storage storage)
        {
            _itemsCount = itemsCount;
            _pickDelay_ms = pickDelay_ms;
            _storage = storage;
        }

        public void Start()
        {
            for (int i = 0; i < _itemsCount; i++)
            {
                _storage.PickItemSemaphore.WaitOne();
                Thread.Sleep(_pickDelay_ms);
                _storage.AccessToStorageSemaphore.WaitOne();

                string item = _storage.PickItem();

                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine($"Pick {item}");
                Console.ForegroundColor = ConsoleColor.Black;

                _storage.AccessToStorageSemaphore.Release();
                _storage.PushItemSemaphore.Release();
            }
        }
    }
}

File: Semaphore_java/src/Customer.java
public class Customer extends Thread{
    private final Integer _itemsCount;
    private final Integer _pushDelay_ms;
    private final Storage _storage;

    public Customer(int itemsCount, int pushDelay_ms, Storage storage) {
        _itemsCount = itemsCount;
        _pushDelay_ms = pushDelay_ms;
        _storage = storage;
    }

    @Override
    public void run(){
        for (Integer i = 0; i < _itemsCount; i++)
        {
            try {
                _storage.PickItemSemaphore.acquire();
                Thread.sleep(_pushDelay_ms);
                _storage.AccessToStorageSemaphore.acquire();

                String item = _storage.pickItem();
                System.out.println("Pick " + item);

                _storage.AccessToStorageSemaphore.release();
                _storage.PushItemSemaphore.release();
            } catch (InterruptedException e) {
                throw new RuntimeException(e);
            }
        }
    }
}